pkg/agent/qrm-plugins/network/staticpolicy/policy_test.go

```diff
@@ -1167,137 +1337,276 @@ func TestGetResourcesAllocation(t *testing.T) {
-			TopologyLevel: pluginapi.TopologyLevel_SOCKET,
-			Annotations: map[string]string{
-				apiconsts.ResourceAnnotationKeyResourceIdentifier: fmt.Sprintf("%s-%s", testEth2NSName, testEth2Name),
-			},
+			description:                     "has no valid nic",
+			hasNic:                          false,
+			expectedAllocatableQuantityList: make([]*pluginapi.TopologyAwareQuantity, 0),
+			expectedCapacityQuantityList:    make([]*pluginapi.TopologyAwareQuantity, 0),
 		},
 	}
-	assert.Equal(t, resp.AllocatableResources[string(apiconsts.ResourceNetBandwidth)].TopologyAwareAllocatableQuantityList, expectedTopologyAwareAllocatableQuantityList)
 
-	expectedTopologyAwareCapacityQuantityList := []*pluginapi.TopologyAwareQuantity{
-		{
-			ResourceValue: float64(21250),
-			Node:          uint64(0),
-			Name:          testEth0Name,
-			Type:          string(apinode.TopologyTypeNIC),
-			TopologyLevel: pluginapi.TopologyLevel_SOCKET,
-			Annotations: map[string]string{
-				// testEth0NSName is empty, so remove the prefix
-				apiconsts.ResourceAnnotationKeyResourceIdentifier: testEth0Name,
+	for _, tc := range testCases {
+		policy := makeStaticPolicy(t, tc.hasNic)
+		assert.NotNil(t, policy)
+
+		resp, err := policy.GetTopologyAwareAllocatableResources(context.TODO(), &pluginapi.GetTopologyAwareAllocatableResourcesRequest{})
+		assert.NotNil(t, resp)
+		assert.NoError(t, err)
+
+		if tc.hasNic {
+			assert.Equal(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].AggregatedAllocatableQuantity, tc.expectedAllocatableQuantityList[0].ResourceValue+tc.expectedAllocatableQuantityList[1].ResourceValue)
+			assert.Equal(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].AggregatedCapacityQuantity, tc.expectedCapacityQuantityList[0].ResourceValue+tc.expectedCapacityQuantityList[1].ResourceValue)
+			assert.Len(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].TopologyAwareAllocatableQuantityList, len(tc.expectedAllocatableQuantityList))
+			assert.Len(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].TopologyAwareCapacityQuantityList, len(tc.expectedCapacityQuantityList))
+
+			assert.Equal(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].TopologyAwareAllocatableQuantityList, tc.expectedAllocatableQuantityList)
+			assert.Equal(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].TopologyAwareCapacityQuantityList, tc.expectedCapacityQuantityList)
+		} else {
+			assert.Equal(t, float64(0), resp.AllocatableResources[string(consts.ResourceNetBandwidth)].AggregatedAllocatableQuantity)
+			assert.Equal(t, float64(0), resp.AllocatableResources[string(consts.ResourceNetBandwidth)].AggregatedCapacityQuantity)
+			assert.Len(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].TopologyAwareAllocatableQuantityList, 0)
+			assert.Len(t, resp.AllocatableResources[string(consts.ResourceNetBandwidth)].TopologyAwareCapacityQuantityList, 0)
+		}
+	}
+
+	/*
```

Comment:
we should delete all the unless codes instead of using comments

Reply:
Oh, it was left by mistake. Just forgot it.

Reply:
Just removed